File: sanghoon/programmers-2/짝지어 제거하기.py
def solution(s):
    stack = []
    for i in range(len(s)):
        if len(stack) == 0 or stack[-1] != s[i]:
            stack.append(s[i])
        else:
            stack.pop()

    return 1 if len(stack) == 0 else 0

s = "baabaa"
solution(s)

File: eunsol/programmers-2/짝지어-제거하기.java
import java.util.Stack;

class Solution {
    public int solution(String s) {
        Stack<String> stack = new Stack<>();
        String[] strArr = s.split("");

        for (String str : strArr) {
            if (!stack.empty() && stack.peek().equals(str)) {
                stack.pop();
            } else {
                stack.push(str);
            }
        }
        return stack.empty() ? 1 : 0;
    }
}